odoo/addons/base/models/ir_ui_view.py

```diff
@@ -357,9 +357,9 @@ def _check_xml(self):
                 for view_arch in view_docs:
                     check = valid_view(view_arch)
                     if not check:
-                        raise ValidationError(_('Invalid view %s definition in %s') % (view.name, view.arch_fs))
+                        raise ValidationError(_('Invalid view %s definition in %s') % (view.xml_id, view.arch_fs))
                     if check == "Warning":
-                        _logger.warning(_('Invalid view %s definition in %s \n%s'), view.name, view.arch_fs, view.arch)
+                        _logger.warning(_('Invalid view %s definition in %s \n%s'), view.xml_id, view.arch_fs, view.arch)
```

Comment:
view_name = view.name if not view.xml_id else '%s (%s)' % (view.name, view.xml_id)
raise ValidationError(_('Invalid view %s definition in %s') % (view_name, view.arch_fs))